src/Core/src/Platform/Windows/MauiWebView.cs

```diff
@@ -143,12 +153,18 @@ void SetupPlatformEvents()
 			NavigationStarting += (sender, args) =>
 			{
 				// Auto map local virtual app dir host, e.g. if navigating back to local site from a link to an external site
-				if (args?.Uri?.ToLowerInvariant().StartsWith(LocalScheme.TrimEnd('/').ToLowerInvariant()) == true)
+				bool isUrlWithLocalScheme = args?.Uri?.ToLowerInvariant().StartsWith(LocalScheme.TrimEnd('/').ToLowerInvariant()) == true;
+				bool isDataUrl = args?.Uri?.StartsWith("data:text/html") ?? false;
```

Comment:
This also needs to pass in a specific `StringComparison` --> probably `Ordinal`.

Reply:
Good point, the data URL check can be improved with checking the Uri object directly. I'll push a commit today.

Reply:
I checked the code, and unfortunately `args?.Uri` is only a `string`, no `Uri`. Do you think it's worthwile to call `Uri.TryParse()` and work with the `Uri` object instead?

Reply:
I think this is discussion is resolved by the later comments.